backend/ibutsu_server/scripts/mongo2postgres.py

```diff
@@ -0,0 +1,296 @@
+#!/bin/env python3
+"""
+Convert a Ibutsu's MongoDB to PSQL
+
+python3 mongo2postgres.py
+    mongodb://localhost/test_artifacts postgresql://ibutsu:ibutsu@localhost:5432/ibutsu -v
+"""
+from argparse import ArgumentParser
+from datetime import datetime
+from datetime import timedelta
+from datetime import timezone
+from uuid import UUID
+
+from bson import ObjectId
+from gridfs import GridFSBucket
+from ibutsu_server.db.models import Artifact
+from ibutsu_server.db.models import Group
+from ibutsu_server.db.models import Project
+from ibutsu_server.db.models import Result
+from ibutsu_server.db.models import Run
+from ibutsu_server.db.models import WidgetConfig
+from pymongo import MongoClient
+from sqlalchemy import create_engine
+from sqlalchemy.ext.declarative import declarative_base
+from sqlalchemy.orm import sessionmaker
+
+
+UUID_1_EPOCH = datetime(1582, 10, 15, tzinfo=timezone.utc)
+UUID_TICKS = 10000000
+UUID_VARIANT_1 = 0b1000000000000000
+ROWS_TO_COMMIT_AT_ONCE = 10000
+MONTHS_TO_KEEP = 3
+MIGRATION_TIME_FRAME = MONTHS_TO_KEEP * 30 * 24 * 60 * 60  # 3 months in seconds
+MIGRATION_LIMIT = 10000000000  # mostly for testing purposes
+
+Base = declarative_base()
+session = None
+
+
+TABLE_MAP = [
+    ("groups", Group),
+    ("projects", Project),
+    ("widgetConfigs", WidgetConfig),
+    ("runs", Run),
+    ("results", Result),
+]
+FILE_MAP = [
+    # only convert artifacts, the reports in the existing DB aren't particularly useful
+    ("fs", Artifact),
+]
+
+# metadata fields that are their own column now
+FIELDS_TO_PROMOTE = [
+    "component",
+    "env",
+]
+# ID fields that must be converted
+ID_FIELDS = [
+    "result_id",
+    "resultId",
+    "run_id",
+    "runId",
+    "project",
+    "run",
+]
+
+# indexes for the tables
+INDEXES = {
+    "results": [
+        "CREATE INDEX results_env ON results(env);",
```

Comment:
Indexes on individual columns can (and should) be created on the models. For example:

```python
class Result(Model):
    env = Column(String, index=True)
```